backport/datasync/datasync.py

```diff
@@ -0,0 +1,335 @@
+import concurrent.futures
+import datetime as dt
+import gzip
+import json
+import os
+import time
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+import rich_click as click
+import structlog
+from botocore.config import Config
+from botocore.exceptions import EndpointConnectionError
+from dataclasses_json import dataclass_json
+from sqlalchemy.engine import Engine
+from sqlmodel import Session
+from tenacity import Retrying
+from tenacity.retry import retry_if_exception_type
+from tenacity.stop import stop_after_attempt
+from tenacity.wait import wait_exponential
+
+from backport.datasync.data_metadata import (
+    variable_data,
+    variable_data_df_from_mysql,
+    variable_data_df_from_s3,
+    variable_metadata,
+)
+from etl import config
+from etl import grapher_model as gm
+from etl.db import get_engine
+from etl.publish import connect_s3
+
+log = structlog.get_logger()
+
+config.enable_bugsnag()
+
+
+S3_BUCKET = "owid-catalog"
+S3_PREFIX = f"baked-variables/{config.DB_NAME}"
```

Comment:
Minor suggestion: Shouldn't we call it simply `variables`? The name `baked-variables` sounds a bit like they have been manipulated.

Reply:
I took this naming from grapher where we use `bakedSite/grapher/variables`. You're right though, it's a bit weird. I already have all the data there and won't be changing it as part of this PR, but I'll reopen this discussion when we start switching URLs in grapher.